packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -282,6 +282,9 @@ export function listClients(context: TCGCContext): SdkClient[] {
   const explicitClients = [...listScopedDecoratorData(context, clientKey)];
   if (explicitClients.length > 0) {
     context.__rawClients = getClientsWithVersioning(context, explicitClients);
+    for (const client of context.__rawClients) {
+      context.arm = isArm(client.service);
+    }
```

Comment:
Here uses the same logic as 
https://github.com/Azure/typespec-azure/blob/45c186059741b396d451099548969a8e1671ae7a/packages/typespec-client-generator-core/src/decorators.ts#L294-L303

Assuming we won't have mixed service (arm and non-arm).